routes/products.js

```diff
@@ -1,32 +1,64 @@
 const express = require('express');
+const { to } = require('await-to-js');
 const router = express.Router();
+
 const models = require('../lib/database/mysql/index');
-const { to } = require('await-to-js');
 const utils = require('../data/utils');
-const Sequelize = require('sequelize');
-const { reviewModel } = require('../lib/database/mysql/index');
+const product_services = require('../services/products');
+const cache = require('../lib/cache/redis');
+const middlwr_caching = require('../data/middlewares/caching_functions');
+
+
+// Get all products using caching
+router.get('/all', middlwr_caching.caching_all_prod ,async(req, res) => {
+    let [err, serv] = await to(product_services.get_all_products());
```

Comment:
First try to fetch from cache. If not in cache then fetch from db and add in cache as well.